streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java

```diff
@@ -0,0 +1,203 @@
+        final MockProcessorContext.CapturedForward capturedForward = context.forwarded().get(0);
+        assertThat(capturedForward.keyValue(), is(new KeyValue<>(key, value)));
+        assertThat(capturedForward.timestamp(), is(timestamp));
+    }
+
+    @Test
+    public void windowedZeroEmitAfterShouldImmediatelyEmit() {
+        final KTableSuppressProcessor<Windowed<String>, Long> processor =
+            new KTableSuppressProcessor<>(getImpl(intermediateEvents(withEmitAfter(ZERO))));
+
+        final MockInternalProcessorContext context = new MockInternalProcessorContext();
+        processor.init(context);
+
+        final long timestamp = arbitraryLong();
+        context.setTimestamp(timestamp);
+        context.setStreamTime(timestamp);
+        final Windowed<String> key = new Windowed<>("hey", arbitraryWindow());
+        final Change<Long> value = new Change<>(arbitraryLong(), arbitraryLong());
+        processor.process(key, value);
+
+        assertThat(context.forwarded(), hasSize(1));
+        final MockProcessorContext.CapturedForward capturedForward = context.forwarded().get(0);
+        assertThat(capturedForward.keyValue(), is(new KeyValue<>(key, value)));
+        assertThat(capturedForward.timestamp(), is(timestamp));
+    }
+
+    @Test
+    public void intermediateSuppressionShouldThrow() {
+        final KTableSuppressProcessor<String, Long> processor =
+            new KTableSuppressProcessor<>(getImpl(intermediateEvents(withEmitAfter(Duration.ofMillis(1)))));
+
+        final MockInternalProcessorContext context = new MockInternalProcessorContext();
+        processor.init(context);
+
+        try {
+            processor.process("hey", new Change<>(null, 1L));
+            fail("expected an exception for now");
+        } catch (final KTableSuppressProcessor.NotImplementedException e) {
+            // expected
+        }
+        assertThat(context.forwarded(), hasSize(0));
+    }
+
+    @Test
+    public void finalResultsSuppressionShouldThrow() {
+        final SuppressedImpl<Windowed<String>, Long> suppress = getImpl(emitFinalResultsOnly(unbounded()));
+        final KTableSuppressProcessor<Windowed<String>, Long> processor =
+            new KTableSuppressProcessor<>(getImpl(SuppressedImpl.buildFinalResultsSuppression(Duration.ofMillis(1), suppress)));
+
+        final MockInternalProcessorContext context = new MockInternalProcessorContext();
+        processor.init(context);
+
+        context.setTimestamp(arbitraryLong());
+        try {
+            processor.process(new Windowed<>("hey", arbitraryWindow()), new Change<>(arbitraryLong(), arbitraryLong()));
+            fail("expected an exception for now");
+        } catch (final KTableSuppressProcessor.NotImplementedException e) {
+            // expected
+        }
+        assertThat(context.forwarded(), hasSize(0));
+    }
+
+    @Test
+    public void finalResultsWith0GraceBeforeWindowEndShouldThrow() {
```

Comment:
Why do we need an extra test for this? Seems to be covered in `finalResultsSuppressionShouldThrow` ?

Reply:
It's a stub of a test that I have in the next PR. When I extracted this one and replaced buffering with exceptions, I just replaced all the method bodies with verification of the exception.